src/main/java/net/datafaker/idnumbers/EnZAIdNumber.java

```diff
@@ -75,9 +74,8 @@ boolean validSsn(String ssn) {
             return false;
         }
 
-        int calculatedChecksum = calculateChecksum(ssn.substring(0, 12));
-        int checksum = Integer.parseInt(ssn.substring(12, 13));
-        return (checksum == calculatedChecksum);
+        int calculatedChecksum = calculateChecksum(ssn, 12);
+        return ssn.charAt(12) - '0' == calculatedChecksum;
```

Comment:
Is there any reason for the intermediate store (assignment)? Why not just combine the two lines?

Reply:
nothing special, could be combined
done